mne/viz/_figure.py

```diff
@@ -1854,6 +1863,56 @@ def _update_yaxis_labels(self):
                    else 'normal')
             text.set_style(sty)
 
+    def _xtick_formatter(self, xval, _=None):
+        """Change the x-axis labels."""
+        if self.mne.time_format == 'datetime':
+            import datetime
+            first_time = self.mne.inst.first_time
+            meas_date = self.mne.inst.info['meas_date']
+            seconds = int(xval + first_time)
+            ms = int((xval + first_time) % 1 * 1e3)
+            if ms == 0:
+                us = int(((xval + first_time) % 1 * 1e6))
```

Comment:
```suggestion
                us = int((xval + first_time) % 1 * 1e6)
```

Reply:
Actually here's a mistake. `%` has precedence over `*`, so you're really calculating the remainder of the division by one, and then multiply by `1e6`. Either use parens or better yet just write `1e6`, i.e. `us = int((xval + first_time) % 1e6)`.

Reply:
Sorry, I didn't see your previous comment - see above.